openpype/tools/publisher/widgets/create_widget.py

```diff
@@ -828,6 +828,7 @@ def _on_create(self):
 
         if success:
             self._set_creator(self._selected_creator)
+            self.variant_input.setText(variant)
```

Comment:
Wasn't sure if that was the best, since the `_set_creator` will reset (remove any uploads for example) the creator and do further checks, so I ended up doing this in case the `_set_creator` is required, and we still trigger the `on_change_variant` i am happy with either option